modules/ROOT/pages/writing.adoc

```diff
@@ -672,7 +678,229 @@ CREATE CONSTRAINT FOR (p:Person) REQUIRE (p.surname) IS UNIQUE
+      .format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("relationship", "MY_REL")
+      .option("relationship.save.strategy", "keys")
+      .option("relationship.source.labels", ":NodeA")
+      .option("relationship.source.save.mode", "Overwrite")
+      .option("relationship.source.node.keys", "idSource:id")
+      .option("relationship.target.labels", ":NodeB")
+      .option("relationship.target.node.keys", "idTarget:id")
+      .option("relationship.target.save.mode", "Overwrite")
+      .option("schema.optimization.relationship.key", true)
+      .option("relationship.keys", "foo,bar")
+      .save()
+----
+
+Under the hood the Spark connector will create the following constraint:
+
+`CREATE CONSTRAINT `spark_RELATIONSHIP_KEY-CONSTRAINT_MY_REL_foo-bar` IF NOT EXISTS FOR ()-[e:MY_REL]->() REQUIRE (e.foo, e.bar) IS RELATIONSHIP KEY`
+
+===== Property type constraints
+
+Since Neo4j 5.11 the database allow to create type constraints for node and relationship properties.
+In oder to leverage this feature we added the option `schema.type.constraint` that will use the DataFrame schema in order to enforce the type.
+Internally the connector will use the following mapping:
+
+.Spark to Cypher constraint type mapping
+|===
+|Spark type |Neo4j Type
+|BooleanType |BOOLEAN
+|StringType |STRING
+|IntegerType |INTEGER
+|LongType |INTEGER
+|FloatType |FLOAT
+|DoubleType |FLOAT
+|DateType |DATE
+|TimestampType |LOCAL DATETIME
+|Custom `pointType` as: Struct { type: string, srid: integer, x: double, y: double, z: double }| POINT
+|Custom `durationType` as: Struct { type: string, months: long, days: long, seconds: long, nanonseconds: integer, value: string }| DURATION
+|DataTypes.createArrayType(BooleanType, false) |LIST<BOOLEAN NOT NULL>
+|DataTypes.createArrayType(StringType, false) |LIST<STRING NOT NULL>
+|DataTypes.createArrayType(IntegerType, false) |LIST<INTEGER NOT NULL>
+|DataTypes.createArrayType(LongType, false) |LIST<INTEGER NOT NULL>
+|DataTypes.createArrayType(FloatType, false) |LIST<FLOAT NOT NULL>
+|DataTypes.createArrayType(DoubleType, false) |LIST<FLOAT NOT NULL>
+|DataTypes.createArrayType(DateType, false) |LIST<DATE NOT NULL>
+|DataTypes.createArrayType(TimestampType, false) |LIST<LOCAL DATETIME NOT NULL>
+|DataTypes.createArrayType(pointType, false) |LIST<POINT NOT NULL>
+|DataTypes.createArrayType(durationType, false) |LIST<DURATION NOT NULL>
+
+|===
+
+For the arrays in particular we use the version without null elements as Neo4j does not allow to have them in arrays.
+
+====== Node Property type constraints
+
+Given the following example:
+
+[source, scala]
+----
+    ds.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Overwrite)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", ":Person:Customer")
+      .option("node.keys", "surname")
+      .option("schema.type.constraint", true)
```

Comment:
Do we expect more settings under `schema.type`? It could just be `schema.type-constraint`, or?
Also, why not `schema.optimization.type.constraint` (or `schema.optimization.type-constraint`) to be more aligned with the others?

Reply:
Can you please check the document? I remember that was something suggested by @ali-ince (but I'm not sure)
Btw `schema.optimization.type.constraint` could be confused with the old `schema.optimization.type`.